### ISSUE TITLE
Convert array indices from size_t to int

### ISSUE THREAD

**Issue by ghost** [KeyWorksRW/ttLib#59]
### Description:
The upside of using **size_t** is that it is a 64-bit value on 64-bit systems. Theoretically, that could result in a performance gain, though in reality it will have little if any effect. An unsigned index makes sense in that you cannot have a negative position into an array.

The advantage of a signed index is that lookup functions can return -1 and the caller doesn't have to cast the result. Instead of

```
size_t pos = cls.LookUp(something);
if (pos != (size_t) -1) {
```

one could simply write:

```
auto pos = cls.LookUp(something);
if (pos >= 0) {
```

If we use a signed value as the index, then the question becomes whether to use **ptrdiff_t** or **int**. Like **size_t**, **ptrdiff_t** is a 64-bit value on 64-bit systems, but again the theoretical performance gain is going to be minimal if at all. As of 2019, both **MSVC** and **CLANG** compilers default to **int** when assigning a constant to an auto variable:

```
for (auto pos = 0;
```

In the above code, _pos_ becomes an **int**.

None of the array classes in **ttLib** are designed to handle 2 million or more entries since each entry is a memory allocation. Using a 32-bit signed value will have no practical limitation.

Bottom line: for classes that contain arrays, use **int** as the index not **size_t**.

### Issue Type:
- [ ] Bug report
- [ ] Feature request
- [ ] Other
